play-json/shared/src/main/scala/Reads.scala

```diff
@@ -129,6 +129,22 @@ object Reads extends ConstraintReads with PathReads with DefaultReads with Gener
   }
 
   implicit val JsArrayReducer = Reducer[JsValue, JsArray](js => JsArray(Array(js)))
+
+  /**
+   * A reads that fails if any of the objects properties are not present in the passed in known properties list.
+   */
+  def failOnUnknownProperties(knownProperties: immutable.Seq[String]): Reads[JsObject] = Reads { jsValue =>
```

Comment:
Yeah I think `Set` makes more sense. Or even `String => Boolean`. Though if we're going to print the allowed properties in the error message being able to control the ordering might be nice.